### ISSUE TITLE
After update to 3.0.3 users cannot post to a group shared calendar with write access checked.

### ISSUE THREAD

**Issue by WNYmathGuy** [nextcloud/calendar#3842]
### Steps to reproduce
1. Have a Nextcloud 22.2.3 instance relying on a MariaDB 10.5.6 database
2. Update to Calendar 3.0.3
3. Have a user make a new calendar entry to the shared by user-group calendar with write access for that group. 
4. Dropdown allows selection of shared calendar but only displays "Personal" calendar as selected.
5. Discover it will only post to the "Personal" default calendar of the user.
6. Try to edit the new entry to move it to the shared calendar and discover it won't move.

### Expected behaviour
It should post the way it did the day before the recent Calendar update was installed.

### Actual behaviour
It only posts to the Personal calendar.

### Calendar app
**Calendar app version:** 3.0.3

**CalDAV-clients used:** via normal web interface of Nextcloud, both with Chrome and Firefox on Ubunty 20.04 Workstation.

### Client configuration
**Browser:** Chrome Version 96.0.4664.110 (Official Build) snap (64-bit), Firefox 95.0.1 (64-bit). 

**Operating system:** Ubuntu 20.04

### Server configuration

**Operating system**: Ubuntu 18.04.6

**Web server:** Apache 2.4.52

**Database:** MariaDB 10.5.6

**PHP version:** 8.0.14

**Nextcloud Version:** 22.2.3
**Updated from an older installed version** System is less than 4 months old.

Link: [https://service.harrogatesquare.com/index.php/s/8MsPBEr55bMW6oC](https://service.harrogatesquare.com/index.php/s/8MsPBEr55bMW6oC)
Password: `xNTYPpotiS`

**Comment by ChristophWurst** [comment]
This might be a regression of https://github.com/nextcloud/calendar/pull/3424.

**Comment by ChristophWurst** [comment]
Or a duplicate of https://github.com/nextcloud/calendar/issues/3838.

**Comment by tdoerschel** [comment]
I can confirm this bug.
But it only happens in the "extended" editor on the right side.
If you use the "simplified editor" (see Settings and Import on bottom left), you can change the calender of the open event.

**Comment by ChristophWurst** [comment]
Thanks for the detail @tdoerschel. I'll make sure someone will try to reproduce and investigate soon.

**Comment by bovender** [comment]
I am experiencing the same problem with the difference that all calendars were created (and shared) by me -- but I'm unable to change the calendar to anything other than the default calendar. Browser console shows no output when I attempt to switch.

**Comment by ChristophWurst** [comment]
It sounds like a regression with the calender picker. I also noticed some weird behavior of the event categories last week. This might be related.

**Comment by szaimen** [comment]
Might be due to https://github.com/nextcloud/calendar/blob/f39e31fec11708970f975dc83e49246decce95dd/src/views/EditSidebar.vue#L95

Shouldn't that be `@select-calendar` instead of `@selectCalendar` ? 

This change was made in https://github.com/nextcloud/calendar/pull/3424

**Comment by dominic-p** [comment]
Just confirming this as well on Windows 10 / Firefox (calendar change works on simple editor, doesn't work on extended editor). Nothing shows up in the console. The selected calendar just doesn't change.

**Comment by ChristophWurst** [comment]
> Shouldn't that be `@select-calendar` instead of `@selectCalendar` ?

Vue should convert this but it's work checking

**Comment by jonathanmmm** [comment]
Can confirm bug, but also doesn't work when only having two not shared calendars (and no shared calendars in the account).
So this bug is not related to group/shared calendars in general (except it is there, if at least one calendar on the whole nextcloud instance is a shared one).
So updating the Title would be good, can somebody confirm in an account without shared/group calendars?
(I confirmed in two seperate nextcloud instances, both nextcloud 22.2.3 and calendar 3.0.3.

**Comment by RalfAlbers** [comment]
I'm having the same issue, but only new permissions seems to bee affected. Older permissions still working fine. Also new created groups are not showing up.
I also ca confirm the workarond with using the simple editor.

**Comment by ChristophWurst** [comment]
Dup of https://github.com/nextcloud/calendar/issues/3838 and fixed by https://github.com/nextcloud/calendar/pull/3846

**Comment by WNYmathGuy** [comment]
Mine appears to be fixed by the update.

**Comment by jonathanmmm** [comment]
> Mine appears to be fixed by the update.

By me, too.

**Comment by jlalu** [comment]
After the update I can move an event to another calendar but I can not move it back to the original calendar.

**Comment by tdoerschel** [comment]
I can confirm @jlalu : With v3.0.4 you can post and change to a shared calendar, but cannot change it back.
However, you can change it to another shared calender.

Also, after a change in calender: when you select the entry again, it still shows the old (before last change) calender in the details. Only after full reload of calender app, the correct details appear.

**Comment by ChristophWurst** [comment]
> After the update I can move an event to another calendar but I can not move it back to the original calendar.

https://github.com/nextcloud/calendar/issues/3325

**Comment by jonathanmmm** [comment]
> > After the update I can move an event to another calendar but I can not move it back to the original calendar.
> 
> https://github.com/nextcloud/calendar/issues/3325

That seems to be it. It even crashed sometimes the whole window (browser asked if I want to leave this page). I can see the event from the original calendar in the trash bin.
Deleting the event in trash bin without reloading doesn't help.
